crates/ruff_linter/src/rules/pydocstyle/rules/indent.rs

```diff
@@ -291,10 +292,58 @@ pub(crate) fn indent(checker: &mut Checker, docstring: &Docstring) {
             let line_indent = leading_space(last);
             let line_indent_size = line_indent.chars().count();
             if line_indent_size > docstring_indent_size {
+                let offset_len = if line_indent.text_len() == last.text_len() {
+                    line_indent_size
+                // The trailing quotes are not on a separate line.
+                } else {
+                    let over_indented_size =
+                        std::cmp::min(line_indent_size - docstring_indent_size, over_indented_size);
+                    match over_indented_size.cmp(&docstring_indent_size) {
+                        // If indentation is exactly the same, no changes are needed.
+                        // Example:
+                        // ```
+                        // def f():
+                        //    """A docstring.
+                        //    Args:
+                        //    ^^^^This line is correctly indented (4 spaces)."""
+                        // ```
+                        Ordering::Equal => {
+                            return;
+                        }
+                        // If the indentation is less then docstring_indent_size,
+                        // replace the over-indentation with `docstring.indentation`.
+                        // Example:
+                        // ```
```

Comment:
The documentation here doesn't match the code's behavior. It isn't returning `docstring.indentation` and 
I'm uncertain if this case is also about over indent, considering that `Ordering::Greater` also handles an over indent.

Reply:
Thanks, I've rephrased the comments. I think it is still an over-indentation case, we return `over_indented_size ` here, which is the result of `line_indent_size - docstring_indent_size` and `line_indent_size > docstring_indent_size`.